src/twisted/application/runner/_pidfile.py

```diff
@@ -0,0 +1,202 @@
+# -*- test-case-name: twisted.application.runner.test.test_pidfile -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+PID file.
+"""
+
+import errno
+from os import getpid, kill
+
+from twisted.python.filepath import IFilePath
+from twisted.logger import Logger
+
+
+
+class PIDFile(object):
+    """
+    PID file.
+
+    Manages a file that remembers a process ID.
+    """
+
+    _log = Logger()
+
+
+    @staticmethod
+    def format(pid=None):
+        """
+        Format a PID file's content.
+
+        @param pid: A process ID.
+        @type pid: int
+
+        @return: Formatted PID file contents.
+        @rtype: L{bytes}
+        """
+        if pid is not None:
+            return u"{}\n".format(pid).encode("utf-8")
+        else:
+            return b""
+
+
+    def __init__(self, filePath):
+        """
+        @param filePath: The path to the PID file on disk.
+        @type filePath: L{IFilePath}
+        """
+        self.filePath = IFilePath(filePath)
+
+
+    def read(self):
+        """
+        Read the process ID stored in this PID file.
+
+        @return: The contained process ID, or C{None} if the PID file does not
```

Comment:
This can be a dangerous API -- users must explicitly test for None, and are likely to not do this in test scenarios. Can we have this API instead raise a custom (or standard) exception like ValueError to indicate "no PID"?